src/network/core/http_curl.cpp

```diff
@@ -12,216 +12,156 @@
 #include "../../stdafx.h"
 #include "../../debug.h"
 #include "../../rev.h"
+#include "../../thread.h"
 #include "../network_internal.h"
 
 #include "http.h"
 
 #include <curl/curl.h>
+#include <mutex>
+#include <condition_variable>
+#include <queue>
 
 #include "../../safeguards.h"
 
 /** Single HTTP request. */
 class NetworkHTTPRequest {
-private:
-	std::string uri;        ///< URI to connect to.
-	HTTPCallback *callback; ///< Callback to send data back on.
-	const char *data;       ///< Data to send, if any.
-
-	CURL *curl = nullptr;          ///< CURL handle.
-	CURLM *multi_handle = nullptr; ///< CURL multi-handle.
-
 public:
-	NetworkHTTPRequest(const std::string &uri, HTTPCallback *callback, const char *data = nullptr);
+	/**
+	 * Create a new HTTP request.
+	 *
+	 * @param uri      the URI to connect to (https://.../..).
+	 * @param callback the callback to send data back on.
+	 * @param data     optionally, the data we want to send. When set, this will be a POST request, otherwise a GET request.
+	 */
+	NetworkHTTPRequest(const std::string &uri, HTTPCallback *callback, const char *data = nullptr) :
+		uri(uri),
+		callback(callback),
+		data(data)
+	{
+	}
 
-	~NetworkHTTPRequest();
+	/**
+	 * Destructor of the HTTP request.
+	 */
+	~NetworkHTTPRequest()
+	{
+		free(this->data);
```

Comment:
I know it was in the old curl too, but where is this memory coming from such that it needs to be freed with `free` ? Can it use new/delete instead? Or better, something automatic?

Reply:
Rubidium asked the same :D

The reason I hold off on that, as that is somewhere inside the network_content code. I have no problems addressing that, but I tried to keep these PRs as low-impact as possible, read: only address the http part.

Next PR will address these things, together with some other bugs and troubles we have, if you don't mind. If you do, let me know :)